File: particle-connect/android/src/main/kotlin/network/particle/connect_plugin/model/FlutterErrorMessage.java
package network.particle.connect_plugin.model;

import com.connect.common.model.ConnectError;
import com.google.gson.annotations.SerializedName;

public class FlutterErrorMessage {
    public static final int ExceptionCode = 10000;
    @SerializedName("code")
    public int code;

    @SerializedName("message")
    public String message;


    public FlutterErrorMessage(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public FlutterErrorMessage() {
    }

    public static FlutterErrorMessage exceptionMsg(String message) {
        return new FlutterErrorMessage(ExceptionCode, message);
    }

    public static FlutterErrorMessage parseConnectError(ConnectError error) {
        return new FlutterErrorMessage(error.getCode(), error.getMessage());
    }
}

File: particle-base/android/src/main/kotlin/network/particle/base_flutter/model/FlutterErrorMessage.kt
package network.particle.base_flutter.model

import androidx.annotation.Keep
import com.google.gson.annotations.SerializedName
@Keep
class FlutterErrorMessage {
    @SerializedName("code")
    var code = 0

    @SerializedName("message")
    var message: String? = null

    constructor(code: Int, message: String?) {
        this.code = code
        this.message = message
    }

    constructor() {}

    companion object {
        const val ExceptionCode = 10000
        fun exceptionMsg(message: String?): FlutterErrorMessage {
            return FlutterErrorMessage(ExceptionCode, message)
        }
    }
}